addons/im_livechat/static/src/embed/cors/store_service_patch.js

```diff
@@ -0,0 +1,28 @@
+import { patch } from "@web/core/utils/patch";
+import { Store } from "@mail/model/store";
+import { Record } from "@mail/model/record";
+import { expirableStorage } from "@im_livechat/embed/common/expirable_storage";
+
+export const GUEST_TOKEN_STORAGE_KEY = "im_livechat_guest_token";
+patch(Store.prototype, {
+    setup() {
+        super.setup(...arguments);
+        this.guest_token = Record.attr(null, {
+            compute() {
+                return expirableStorage.getItem(GUEST_TOKEN_STORAGE_KEY);
+            },
+            onUpdate() {
+                if (this.guest_token) {
+                    expirableStorage.setItem(GUEST_TOKEN_STORAGE_KEY, this.guest_token);
+                    this.store.env.services.bus_service.addChannel(
+                        `mail.guest_${this.guest_token}`
+                    );
+                    return;
+                }
+                expirableStorage.removeItem(GUEST_TOKEN_STORAGE_KEY);
+                this.store.env.services.bus_service.deleteChannel(`mail.guest_${this.guest_token}`);
+            },
+            eager: true,
+        });
+    },
+});
```

Comment:
This works on page load, but the immediate notify to other tab from local storage `setItem` is missing.
This problem was also affecting other similar pattern, with this PR: https://github.com/odoo/odoo/pull/200011
Basically we need a `onStorage` to observe when the local storage value is changed.

Reply:
I assumed it would work with the SharedWorker since the main tab already adds the guest channel but you are right for the WebWorker case, I'll add the `onStorage`.

Reply:
most probably comes from [this](https://github.com/odoo/odoo/pull/199779/files#diff-86b2eff8d0ed33134982391adf6c9efa5412e2da164550c728dfbd60ce14edc7R199) that was already fixed [here](https://github.com/odoo/odoo/pull/200531). Since the thread is not considered as loaded, the messages are not displayed (even though received on the bus)